docs/release/controllers.md

```diff
@@ -0,0 +1,151 @@
+
+The Flux controllers and their Go API packages are released by following the
+[Go module version numbering](https://go.dev/doc/modules/version-numbers) conventions:
+
+- `vX.Y.Z-RC.W` release candidates e.g. `v1.0.0-RC.1`
+- `vX.Y.Z` stable releases e.g. `v1.0.0`
+
+The API versioning and controller versioning are indirectly related. For example,
+a source-controller minor release `v1.1.0` can introduce a new API version
+`v1beta1` for a Kind `XRepository` in the `source.toolkit.fluxcd.io` group.
+
+### Release candidates
+
+Release candidates are intended for testing new features or improvements before a final release.
+
+In most cases, a maintainer will publish a release candidate of a controller for Flux users
+to tests it on their staging clusters. Release candidates are not meant to be deployed in production
+unless advised to do so by a maintainer.
+
+### Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security
+vulnerabilities or fixes to severe problems with no workaround.
+
+Patch releases do not contain breaking changes, feature additions or any type of user-facing changes.
+If a CVE fix requires a breaking change, then a minor release will provide the fix.
+
+We expect users to be running the latest patch release of a given minor release as soon as the
+controller release is included in a Flux patch release.
+
+### Minor releases
+
+Minor releases are intended for backwards compatible feature additions and improvements.
+Note that breaking changes may occur if required by a security vulnerability fix.
+
+In addition, minor releases are used when updating Kubernetes dependencies such
+as `k8s.io/api` from one minor version to another.
+
+In effect, this means a new minor version will at least be released for all Flux
+controllers approximately every four months, following each Kubernetes minor version release.
+To properly validate the controllers against the latest Kubernetes version,
+we typically allocate a time window of around two weeks for end-to-end testing of Flux controllers.
+
+It is worth noting that in certain scenarios where project dependencies are not in sync with
+the Kubernetes version or conflicts arise, this two-week timeframe may prove insufficient,
+requiring additional time to address the issues appropriately.
+
+### Major releases
+
+Major releases are intended for drastic changes in the controller behaviour or security stance.
+
+A controller major release will be announced ahead of time throughout all communication channels,
+and a support window of one year will be provided for the previous major version.
+
+## Release cadence
+
+Flux controllers are at least released at the same rate as Kubernetes, following their cadence of three
+minor releases per year. After each Kubernetes minor release, all controllers are tested against the latest
+Kubernetes version and are released at approximately two weeks after Kubernetes.
```

Comment:
Not relevant to this PR getting merged or not but we have to make sure that we have a process in place that ensures we're following what is stated here. At the moment we have no formal release management or calendar or anything like that in place so as it is now it would depend on someone recalling to do this after each K8s release.